packages/server-wallet/src/handlers/join-channel.ts

```diff
@@ -53,17 +53,19 @@ export function joinChannel(
   channelState: ChannelState
 ): HandlerResult {
   // TODO: use Action creator from another branch
-  const signStateAction = (sv: StateVariables): SignState => ({
-    type: 'SignState',
-    channelId: args.channelId,
-    ...sv,
-  });
+  const signStateAction = (sv: StateVariables): SignState => {
+    return {
+      type: 'SignState',
+      channelId: args.channelId,
+      ...sv,
```

Comment:
nit: it seems dangerous (but probably innocuous here) to spread `sv` at the end.

```suggestion
      ...sv,
      type: 'SignState',
      channelId: args.channelId,
```